src/http/http_response.f90

```diff
@@ -36,6 +39,29 @@ subroutine append_header(this, key, value)
         end if
     
     end subroutine append_header
+
+    function get_header_value(this, find_key) result(retval)
+        class(response_type), intent(in) :: this
+        character(*), intent(in) :: find_key
+        character(:), allocatable :: retval
+        type(string_type) :: temp, h_key
+        type(header_type), allocatable :: header(:)
+        integer :: i
+        
+        header = this%header
```

Comment:
You only use `header` to refer to it on the RHS, so here you're creating an unnecessary copy. I suggest removing the use of `header` and just referring to it as `this % header`.